tests/dbshell/test_oracle.py

```diff
@@ -1,33 +1,46 @@
+from subprocess import CalledProcessError
 from unittest import mock, skipUnless
 
 from django.db import connection
-from django.db.backends.oracle.client import DatabaseClient
 from django.test import SimpleTestCase
 
 
+# Skip if we cannot expect cx_Oracle to be available.
 @skipUnless(connection.vendor == 'oracle', 'Oracle tests')
 class OracleDbshellTests(SimpleTestCase):
-    def _run_dbshell(self, rlwrap=False):
-        """Run runshell command and capture its arguments."""
-        def _mock_subprocess_call(*args):
-            self.subprocess_args = tuple(*args)
-            return 0
 
-        client = DatabaseClient(connection)
-        self.subprocess_args = None
-        with mock.patch('subprocess.call', new=_mock_subprocess_call):
-            with mock.patch('shutil.which', return_value='/usr/bin/rlwrap' if rlwrap else None):
-                client.runshell()
-        return self.subprocess_args
+    def setUp(self):
```

Comment:
Please revert all (mostly unrelated) changes from the `tests/dbshell/test_oracle.py`. Added test (`test_fail_exec`) checks only arguments passed to the `sqlplus`. You can get the same by amending format in the test added in the original patch i.e. `tests.backends.oracle.tests.TransactionalTests.test_password_with_at_sign`.

Reply:
Changed it to just update the one test you mentioned. test_fail_exec was also testing that errors are propagated correctly from subprocess.call but I agree it's orthogonal.